lib/stove-testing-e2e-wiremock/src/test/kotlin/com/trendyol/stove/testing/e2e/wiremock/WireMockOperationsTest.kt

```diff
@@ -0,0 +1,548 @@
+
+class WireMockOperationsTest :
+  FunSpec({
+
+    /**
+     * Configures a POST request mock using [WireMockSystem.mockPostConfigure].
+     *
+     * @param urlMatcher A [UrlPattern] used to match the request URL. Defaults to [urlEqualTo] with the provided [url].
+     *
+     * This test demonstrates how to use the default URL pattern for [WireMockSystem.mockPostConfigure].
+     */
+    test("Wiremock mockPostConfigure should mock urls with urlEqualTo(url) pattern in default") {
+      val url = "/post-url"
+      val client = HttpClient.newBuilder().build()
+      val reqBuilder = HttpRequest
+        .newBuilder(URI("http://localhost:9098/$url"))
+        .header("Content-Type", "application/json")
+
+      TestSystem.validate {
+        wiremock {
+          mockPostConfigure("/$url") { req, _ ->
+            req
+              .withRequestBody(equalTo("request2"))
+              .withHeader("Content-Type", ContainsPattern("application/json"))
+              .willReturn(
+                aResponse()
+                  .withBody("response2")
+                  .withStatus(200)
+                  .withHeader("Content-Type", "application/json; charset=UTF-8")
+              )
+          }
+          mockPostConfigure("/$url") { req, _ ->
+            req
+              .withRequestBody(equalTo("request1"))
+              .withHeader("Content-Type", ContainsPattern("application/json"))
+              .willReturn(
+                aResponse()
+                  .withBody("response1")
+                  .withStatus(200)
+                  .withHeader("Content-Type", "application/json; charset=UTF-8")
+              )
+          }
+        }
+      }
+
+      withContext(Dispatchers.IO) {
+        val request2 = reqBuilder.POST(BodyPublishers.ofString("request2")).build()
+        val response2 = client.send(request2, BodyHandlers.ofString())
+        response2.body() shouldBe "response2"
+
+        val request1 = reqBuilder.POST(BodyPublishers.ofString("request1")).build()
+        val response1 = client.send(request1, BodyHandlers.ofString())
+        response1.body() shouldBe "response1"
+      }
+    }
+
+    /**
+     * Configures a POST request mock using [WireMockSystem.mockPostConfigure].
+     *
+     * @param urlMatcher A [UrlPattern] used to match the request URL. Defaults to [urlEqualTo] with the provided [url].
+     *
+     * This test demonstrates how to configure and override the default URL matcher for [WireMockSystem.mockPostConfigure].
+     */
+    test("Wiremock mockPostConfigure should accept overridden urlMatcher") {
+      val url = "categories/createCategory"
+      val client = HttpClient.newBuilder().build()
+      val reqBuilder = HttpRequest
+        .newBuilder(URI("http://localhost:9098/$url"))
+        .header("Content-Type", "application/json")
+
+      TestSystem.validate {
+        wiremock {
+          mockPostConfigure("/$url", { urlPathMatching("/categories/.*") }) { req, _ ->
```

Comment:
The place where you define the URL should be vice versa.

meaning:

`mockPostConfigure("/categories/.*", { urlPathMatching(it) }) {` becase you have the `it` reference for the url already and you're making a choice to change the matching strategy.